src/coreclr/vm/wasm/helpers.cpp

```diff
@@ -0,0 +1,479 @@
+}
+
+EXTERN_C FCDECL2(Object*, RhpNewVariableSizeObject, CORINFO_CLASS_HANDLE typeHnd_, INT_PTR size)
+{
+    PORTABILITY_ASSERT("RhpNewVariableSizeObject is not implemented on wasm");
+    return nullptr;
+}
+
+EXTERN_C FCDECL1(Object*, RhpNewMaybeFrozen, CORINFO_CLASS_HANDLE typeHnd_)
+{
+    PORTABILITY_ASSERT("RhpNewMaybeFrozen is not implemented on wasm");
+    return nullptr;
+}
+
+EXTERN_C FCDECL2(Object*, RhpNewArrayFast, CORINFO_CLASS_HANDLE typeHnd_, INT_PTR size)
+{
+    PORTABILITY_ASSERT("RhpNewArrayFast is not implemented on wasm");
+    return nullptr;
+}
+
+EXTERN_C FCDECL2(Object*, RhpNewPtrArrayFast, CORINFO_CLASS_HANDLE typeHnd_, INT_PTR size)
+{
+    PORTABILITY_ASSERT("RhpNewPtrArrayFast is not implemented on wasm");
+    return nullptr;
+}
+
+EXTERN_C FCDECL2(Object*, RhpNewArrayFastAlign8, CORINFO_CLASS_HANDLE typeHnd_, INT_PTR size)
+{
+    PORTABILITY_ASSERT("RhpNewArrayFastAlign8 is not implemented on wasm");
+    return nullptr;
+}
+
+EXTERN_C FCDECL1(Object*, RhpNewFastAlign8, CORINFO_CLASS_HANDLE typeHnd_)
+{
+    PORTABILITY_ASSERT("RhpNewFastAlign8 is not implemented on wasm");
+    return nullptr;
+}
+
+EXTERN_C FCDECL1(Object*, RhpNewFastMisalign, CORINFO_CLASS_HANDLE typeHnd_)
+{
+    PORTABILITY_ASSERT("RhpNewFastMisalign is not implemented on wasm");
+    return nullptr;
+}
+
+EXTERN_C FCDECL1(Object*, RhpNewFast, CORINFO_CLASS_HANDLE typeHnd_)
+{
+    PORTABILITY_ASSERT("RhpNewFast is not implemented on wasm");
+    return nullptr;
+}
+
+EXTERN_C FCDECL1(Object*, RhpNew, CORINFO_CLASS_HANDLE typeHnd_)
+{
+    PORTABILITY_ASSERT("RhpNew is not implemented on wasm");
+    return nullptr;
+}
+
+EXTERN_C FCDECL2(Object*, RhpNewArrayMaybeFrozen, CORINFO_CLASS_HANDLE typeHnd_, INT_PTR size)
+{
+    PORTABILITY_ASSERT("RhpNewArrayMaybeFrozen is not implemented on wasm");
+    return nullptr;
+}
+
+EXTERN_C FCDECL2(Object*, RhNewString, CORINFO_CLASS_HANDLE typeHnd_, DWORD stringLength)
+{
+    PORTABILITY_ASSERT("RhNewString is not implemented on wasm");
+    return nullptr;
+}
+
+extern "C" void STDCALL ThePreStubPatchLabel(void)
+{
+    PORTABILITY_ASSERT("ThePreStubPatchLabel is not implemented on wasm");
+}
+
+LONG CLRNoCatchHandler(EXCEPTION_POINTERS* pExceptionInfo, PVOID pv)
+{
+    PORTABILITY_ASSERT("CLRNoCatchHandler is not implemented on wasm");
+    return EXCEPTION_CONTINUE_SEARCH;
+}
+
+EXTERN_C void STDMETHODCALLTYPE ProfileEnterNaked(FunctionIDOrClientID functionIDOrClientID)
+{
+    PORTABILITY_ASSERT("ProfileEnterNaked is not implemented on wasm");
+}
+
+EXTERN_C void STDMETHODCALLTYPE ProfileLeaveNaked(UINT_PTR clientData)
+{
+    PORTABILITY_ASSERT("ProfileLeaveNaked is not implemented on wasm");
+}
+
+EXTERN_C void STDMETHODCALLTYPE ProfileTailcallNaked(UINT_PTR clientData)
+{
+    PORTABILITY_ASSERT("ProfileTailcallNaked is not implemented on wasm");
+}
+
+void InitJITWriteBarrierHelpers()
+{
+    // TODO: implement me
```

Comment:
I'd like to suggest that wasm specific fixmes and todos be something like `WASM-TODO` so they're easy to find. We've been doing this somewhat consistently for the interpreter too.